engine/util/src/main/scala/pl/touk/nussknacker/engine/util/MathUtils.scala

```diff
@@ -0,0 +1,114 @@
+    withNotNullValues(n1, n2) {
+      withValuesWithTheSameType(n1, n2)(new SameNumericTypeHandler {
+        override def onBytes(n1: Byte, n2: Byte): Byte = Math.max(n1, n2).byteValue()
+        override def onShorts(n1: Short, n2: Short): Short = Math.max(n1, n2).shortValue()
+        override def onInts(n1: Int, n2: Int): Int = Math.max(n1, n2)
+        override def onLongs(n1: Long, n2: Long): Long = Math.max(n1, n2)
+        override def onBigIntegers(n1: BigInteger, n2: BigInteger): BigInteger = n1.max(n2)
+        override def onFloats(n1: Float, n2: Float): Float = Math.max(n1, n2)
+        override def onDoubles(n1: Double, n2: Double): Double = Math.max(n1, n2)
+        override def onBigDecimals(n1: java.math.BigDecimal, n2: java.math.BigDecimal): java.math.BigDecimal = n1.max(n2)
+      })
+    }
+  }
+
+  def min(n1: Number, n2: Number): Number = {
+    implicit val promotionStrategy: NumberTypesPromotionStrategy = NumberTypesPromotionStrategy.ForMinMax
+    withNotNullValues(n1, n2) {
+      withValuesWithTheSameType(n1, n2)(new SameNumericTypeHandler {
+        override def onBytes(n1: Byte, n2: Byte): Byte = Math.min(n1, n2).byteValue()
+        override def onShorts(n1: Short, n2: Short): Short = Math.min(n1, n2).shortValue()
+        override def onInts(n1: Int, n2: Int): Int = Math.min(n1, n2)
+        override def onLongs(n1: Long, n2: Long): Long = Math.min(n1, n2)
+        override def onBigIntegers(n1: BigInteger, n2: BigInteger): BigInteger = n1.min(n2)
+        override def onFloats(n1: Float, n2: Float): Float = Math.min(n1, n2)
+        override def onDoubles(n1: Double, n2: Double): Double = Math.min(n1, n2)
+        override def onBigDecimals(n1: java.math.BigDecimal, n2: java.math.BigDecimal): java.math.BigDecimal = n1.min(n2)
+      })
+    }
+  }
+
+  def sum(n1: Number, n2: Number): Number = {
+    implicit val promotionStrategy: NumberTypesPromotionStrategy = NumberTypesPromotionStrategy.ForMathOperation
+    withNotNullValues(n1, n2) {
+      withValuesWithTheSameType(n1, n2)(new SameNumericTypeHandler {
+        override def onBytes(n1: Byte, n2: Byte): Byte = throw new IllegalStateException("Bytes should be promoted to Ints before addition")
+        override def onShorts(n1: Short, n2: Short): Short = throw new IllegalStateException("Shorts should be promoted to Ints before addition")
+        override def onInts(n1: Int, n2: Int): Int = n1 + n2
+        override def onLongs(n1: Long, n2: Long): Long = n1 + n2
+        override def onBigIntegers(n1: BigInteger, n2: BigInteger): BigInteger = n1.add(n2)
+        override def onFloats(n1: Float, n2: Float): Float = n1 + n2
+        override def onDoubles(n1: Double, n2: Double): Double = n1 + n2
+        override def onBigDecimals(n1: java.math.BigDecimal, n2: java.math.BigDecimal): java.math.BigDecimal = n1.add(n2)
+      })
+    }
+  }
+  
+  private def withNotNullValues(n1: Number, n2: Number)(f: => Number)
```

Comment:
Maybe we can make this (and withValuesWithTheSameType) protected? So that more cases can be added in models if needed (e.g. multiplication or sth else?)